### PR TITLE
Update to TileDB 2.10 and Support TILEDB_BOOL

### PR DESCRIPTION
**Pull request by Shelnutt2**
Update to TileDB 2.10 and Support TILEDB_BOOL datatype.

### CODE REVIEW

**Comment by snagles** [comment]
Seems like commits/titles aren't in sync with the change? Support for tiledb_bool or tiledb_blob?

**Review by antalakas — COMMENTED** [review]
Is this about TILEDB_BLOB or TILEDB_BOOL?

File: query.go [hunk 1]
```diff
@@ -1644,6 +1649,11 @@ func (q *Query) BufferNullable(attributeOrDimension string) (interface{}, []uint
 		length := (*cbufferSize) / C.sizeof_int32_t
 		buffer = (*[1 << 46]C.int8_t)(cbuffer)[:length:length]
 
+	case TILEDB_BOOL:
+		ret = C.tiledb_query_get_buffer(q.context.tiledbContext, q.tiledbQuery, cAttributeOrDimension, &cbuffer, &cbufferSize)
```
Comment (thetorpedodog) [inline query.go:1653]:
```suggestion
```

File: query.go [hunk 2]
```diff
@@ -2133,6 +2154,17 @@ func (q *Query) SetBufferVarNullable(attributeOrDimension string, offset []uint6
 		// Store slice so underlying array is not gc'ed
 		q.buffers = append(q.buffers, tmpBuffer)
 		cbuffer = unsafe.Pointer(&(tmpBuffer)[0])
+
+	case reflect.Bool:
+		// Set buffersize
+		bufferSize = bufferSize * bytesizes.Float64
```
Comment (thetorpedodog) [inline query.go:2160]:
```suggestion
		bufferSize = bufferSize * bytesizes.Bool
```

File: query.go [hunk 3]
```diff
@@ -1424,7 +1424,7 @@ func (q *Query) Buffer(attributeOrDimension string) (interface{}, error) {
 		length := (*cbufferSize) / C.sizeof_int64_t
 		buffer = (*[1 << 46]int64)(cbuffer)[:length:length]
 
-	case TILEDB_UINT8:
+	case TILEDB_UINT8, TILEDB_BLOB:
```
Comment (thetorpedodog) [inline query.go:1427]: Is this an intentional related change?

**Review by thetorpedodog — CHANGES_REQUESTED** [review]

File: query.go (hunk 3 above)
Comment (thetorpedodog) [reply query.go:1427]: It's fine that it's here, just wanted to make sure there wasn't other stuff that was missing for it to work.

**Review by thetorpedodog — APPROVED** [review]
With the fix I just applied, this should be ready to go in. So Seth (or anybody else), if you’re OK with my change, merge it.
